python/cudf/bindings/parquet.pyx

```diff
@@ -37,76 +33,60 @@ cpdef cpp_read_parquet(path, columns=None, row_group=None, skip_rows=None,
     cudf.io.parquet.to_parquet
     """
 
-    # Setup arguments
-    cdef pq_read_arg pq_reader = pq_read_arg()
+    # Setup reader options
+    cdef ParquetReaderOptions options = ParquetReaderOptions()
+    for col in columns or []:
+        options.columns.push_back(str(col).encode())
+    options.strings_to_categorical = strings_to_categorical
 
+    # Create reader from source
     if not os.path.isfile(path) or not os.path.exists(path):
         raise FileNotFoundError(
             errno.ENOENT, os.strerror(errno.ENOENT), path
         )
-    path = str(path).encode()
-    source_ptr = <char*>path
-    pq_reader.source_type = FILE_PATH
-    pq_reader.source = source_ptr
-
-    usecols = columns
-    arr_cols = []
-    pq_reader.use_cols_len = 0
-    cdef vector[const char*] vector_use_cols
-    if usecols is not None:
-        for col in usecols:
-            arr_cols.append(str(col).encode())
-        pq_reader.use_cols_len = len(usecols)
-
-    vector_use_cols = arr_cols
-    pq_reader.use_cols = vector_use_cols.data()
-
-    if row_group is not None:
-        pq_reader.row_group = row_group
-    else:
-        pq_reader.row_group = -1
+    cdef unique_ptr[ParquetReader] reader
+    reader = unique_ptr[ParquetReader](
+        new ParquetReader(str(path).encode(), options)
+    )
 
+    # Read data into columns
+    cdef cudf_table table
     if skip_rows is not None:
-        pq_reader.skip_rows = skip_rows
-    else:
-        pq_reader.skip_rows = 0
-
-    if num_rows is not None:
-        pq_reader.num_rows = num_rows
+        table = reader.get().read_rows(
+            skip_rows,
+            num_rows if num_rows is not None else 0
+        )
+    elif num_rows is not None:
+        table = reader.get().read_rows(
+            skip_rows if skip_rows is not None else 0,
+            num_rows
+        )
+    elif row_group is not None:
+        table = reader.get().read_row_group(row_group)
     else:
-        pq_reader.num_rows = -1
-
-    pq_reader.strings_to_categorical = strings_to_categorical
-
-    # Call read_parquet
-    with nogil:
-        result = read_parquet(&pq_reader)
-
-    check_gdf_error(result)
-
-    out = pq_reader.data
-    if out is NULL:
-        raise ValueError("Failed to parse Parquet")
+        table = reader.get().read_all()
 
-    # Extract parsed columns
+    # Extract read columns
     outcols = []
     new_names = []
-    for i in range(pq_reader.num_cols_out):
-        data_mem, mask_mem = gdf_column_to_column_mem(out[i])
+    cdef gdf_column* column
+    for i in range(table.num_columns()):
+        column = table.get_column(i)
+        data_mem, mask_mem = gdf_column_to_column_mem(column)
         outcols.append(Column.from_mem_views(data_mem, mask_mem))
-        new_names.append(out[i].col_name.decode())
-        free(out[i].col_name)
-        free(out[i])
+        new_names.append(column.col_name.decode())
+        free(column.col_name)
+        free(column)
```

Comment:
```suggestion
```
Hrmmm... I don't think this is need either. `cudf::table` internally is a vector and `get_column(i)` just returns that element, without allocating a new pointer.
It's from previous code where columns were malloc'ed; also need to change CSV reader where this was from... @vuule

Reply:
Brain fart, we do, since the columns are still allocated from libcudf, and this was just vector of pointers. D'oh.